src/main/java/com/gym/dao/TraineeDao.java

```
@@ -0,0 +1,58 @@
+package com.gym.dao;
+
+import com.gym.model.Trainee;
+import com.gym.storage.Storage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.Optional;
+
+@Slf4j
+@Repository
+public class TraineeDao {
+    private Storage storage;
+
+    @Autowired
+    public void setStorage(Storage storage) {
+        this.storage = storage;
+    }
+
+    public void save(Trainee trainee) {
+        storage.getTrainees().put(trainee.getId(), trainee);
+        log.info("Saved trainee: {}", trainee);
+    }
+
+    public Trainee update(Trainee trainee) {
+        if (storage.getTrainees().containsKey(trainee.getId())) {
+            storage.getTrainees().put(trainee.getId(), trainee);
+            log.info("Updated trainee: {}", trainee);
+            return trainee;
+        } else {
+            log.warn("Trainee with id {} not found for update", trainee.getId());
+            return null;
```

it is perfect you added logic of user checking before update. But lets throw exception instead of returning null.
IlligalArgumentException would be good